File: games/csharp_games/LostInSpace/Planet.cs
using SplashKitSDK;

namespace LostInSpace
{
    // Planet kind enum
    enum PlanetKind
    {
        MERCURY,
        VENUS,
        EARTH,
        MARS,
        JUPITER,
        SATURN,
        URANUS,
        NEPTUNE,
        PLUTO
    }

    public class Planet
    {
        // Constants
        const float PLANET_SPEED = 0.1f;
        const float PLANET_ROTATE = 0.05f;

        // Fields
        private Sprite _planetSprite;
        private bool _visited;

        // Constructor
        public Planet(int x, int y)
        {
            //Set visited property to false, by default
            _visited = false;

            //Randomly set a new planets sprite
            _planetSprite = new Sprite(_PlanetBitmap((PlanetKind)SplashKit.Rnd(9)));

            //Set planet x and y
            _planetSprite.X = (x - _planetSprite.Width) / 2;
            _planetSprite.Y = (y - _planetSprite.Height) / 2;

            //Set planet velocity and rotation
            _planetSprite.Dx = PLANET_SPEED;
            _planetSprite.Rotation = PLANET_ROTATE;
        }

        // Read-only property to return the planet sprite
        public Sprite Sprite
        {
            get { return _planetSprite; }
        }

        // Property for whether the planet has been visited or not
        public bool Visited
        {
            get { return _visited; }
            set { _visited = value; }
        }

        /**
        * -----------------------
        * Public Methods
        * -----------------------
        */

        // Draw the planet sprite and a circle around if visited
        public void Draw()
        {
            _planetSprite.Draw();

            if (_visited)
            {
                Circle c = _planetSprite.Circle();
                c.Radius += 10;
                SplashKit.DrawCircle(Color.BrightGreen, c);
            }
        }

        // Update the planet sprite
        public void Update()
        {
            _planetSprite.Update();
        }

        /**
        * -----------------------
        * Private Methods
        * -----------------------
        */

        // Return Bitmap from Planet kind
        private Bitmap _PlanetBitmap(PlanetKind kind)
        {
            switch (kind)
            {
                case PlanetKind.MERCURY:
                    return SplashKit.BitmapNamed("mercury");
                case PlanetKind.VENUS:
                    return SplashKit.BitmapNamed("venus");
                case PlanetKind.EARTH:
                    return SplashKit.BitmapNamed("earth");
                case PlanetKind.MARS:
                    return SplashKit.BitmapNamed("mars");
                case PlanetKind.JUPITER:
                    return SplashKit.BitmapNamed("jupiter");
                case PlanetKind.SATURN:
                    return SplashKit.BitmapNamed("saturn");
                case PlanetKind.URANUS:
                    return SplashKit.BitmapNamed("uranus");
                case PlanetKind.NEPTUNE:
                    return SplashKit.BitmapNamed("neptune");
                case PlanetKind.PLUTO:
                    return SplashKit.BitmapNamed("pluto");
                default:
                    return SplashKit.BitmapNamed("mercury");
            }
        }
    }
}

File: games/cpp_games/LostInSpace-TeamName/planet.cpp
#include "splashkit.h"
#include "planet.h"

#define PLANET_SPEED 0.1
#define PLANET_ROTATE 0.05

bitmap planet_bitmap(planet_kind kind)
{
    switch (kind)
    {
        case MERCURY:
            return bitmap_named("mercury");
        case VENUS:
            return bitmap_named("venus");
        case EARTH:
            return bitmap_named("earth");
        case MARS:
            return bitmap_named("mars");
        case JUPITER:
            return bitmap_named("jupiter");
        case SATURN:
            return bitmap_named("saturn");
        case URANUS:
            return bitmap_named("uranus");
        case NEPTUNE:
            return bitmap_named("neptune");
        case PLUTO:
            return bitmap_named("pluto");
        default:
            return bitmap_named("mercury");
    }
}

planet_data new_planet(double x, double y)
{
    planet_data result;

    //Set visited property to false, by default
    result.visited = false;

    bitmap default_bitmap = planet_bitmap(MERCURY);

    //Randomly set a new planets sprite
    result.planet_sprite = create_sprite(planet_bitmap(static_cast<planet_kind>(rnd(9))));

    //Set planet x and y
    sprite_set_x(result.planet_sprite, x - sprite_width(result.planet_sprite)/2);
    sprite_set_y(result.planet_sprite, y - sprite_height(result.planet_sprite)/2);

    //Set planet velocity and rotation
    sprite_set_dx(result.planet_sprite, PLANET_SPEED);
    sprite_set_rotation(result.planet_sprite, PLANET_ROTATE);

    return result;
}

void draw_planet(const planet_data &planet)
{
    draw_sprite(planet.planet_sprite);

    if (planet.visited)
    {
        circle c = sprite_circle(planet.planet_sprite);
        c.radius += 10;
        draw_circle(COLOR_BRIGHT_GREEN, c);
    }
}

void update_planet(planet_data &planet_to_update)
{
    // Apply movement based on rotation and velocity in the sprite
    update_sprite(planet_to_update.planet_sprite);
}